File: movil/lib/models/taller.dart
class Taller {
  final int idTaller;
  final int duenoId;
  final String nombreDueno;
  final String emailDueno;
  final String telefonoDueno;
  final String nombreTaller;
  final String direccion;
  final String nit;
  final double latitudDecimal;
  final double longitudDecimal;

  Taller({
    required this.idTaller,
    required this.duenoId,
    required this.nombreDueno,
    required this.emailDueno,
    required this.telefonoDueno,
    required this.nombreTaller,
    required this.direccion,
    required this.nit,
    required this.latitudDecimal,
    required this.longitudDecimal,
  });

  factory Taller.fromJson(Map<String, dynamic> json) => Taller(
        idTaller: json['id_taller'],
        duenoId: json['dueño_id'],
        nombreDueno: json['nombre_dueno'] ?? '',
        emailDueno: json['email_dueno'] ?? '',
        telefonoDueno: json['telefono_dueno'] ?? '',
        nombreTaller: json['nombre_taller'] ?? '',
        direccion: json['direccion'] ?? '',
        nit: json['nit'] ?? '',
        latitudDecimal: double.tryParse(json['latitud_decimal'].toString()) ?? 0,
        longitudDecimal: double.tryParse(json['longitud_decimal'].toString()) ?? 0,
      );
}

File: backend/app/schemas/taller.py
# ============================================================
# schemas/taller.py
#
# CU3: Gestión de Talleres
#
from pydantic import BaseModel, EmailStr
from typing import Optional
from decimal import Decimal


# -------------------------------------------------------
# CU3 - REGISTRAR TALLER
# Body que el administrador envía para crear un taller.
# También crea el usuario asociado con rol='taller'.
# -------------------------------------------------------
class TallerCreate(BaseModel):
    # Datos del usuario dueño (se crea junto con el taller)
    nombre_dueno:      str
    email:             EmailStr
    password:          str
    telefono:          Optional[str] = None

    # Datos propios del taller
    nombre_taller:     str
    direccion:         Optional[str] = None
    nit:               Optional[str] = None

    # Coordenadas geográficas (usadas en CU11 - asignación inteligente)
    latitud_decimal:   Optional[Decimal] = None
    longitud_decimal:  Optional[Decimal] = None
    tenant_id:         Optional[int] = 1


# -------------------------------------------------------
# CU3 - ACTUALIZAR TALLER (parcial)
# Todos los campos son opcionales.
# Solo se actualizan los campos que vengan en el body.
# -------------------------------------------------------
class TallerUpdate(BaseModel):
    nombre_taller:     Optional[str]     = None
    direccion:         Optional[str]     = None
    nit:               Optional[str]     = None
    latitud_decimal:   Optional[Decimal] = None
    longitud_decimal:  Optional[Decimal] = None

    # También se puede actualizar info del dueño
    nombre_dueno:      Optional[str]     = None
    telefono:          Optional[str]     = None
    tenant_id:         Optional[int]     = None


# -------------------------------------------------------
# CU3 - RESPUESTA TALLER
# Lo que se devuelve al consultar un taller.
# Incluye datos del dueño embebidos para que Angular
# no tenga que hacer una segunda llamada.
# Nunca se incluye password_hash.
# -------------------------------------------------------
class TallerOut(BaseModel):
    id_taller:         int
    dueño_id:          int

    # Datos del dueño (del usuario relacionado)
    nombre_dueno:      str
    email_dueno:       str
    telefono_dueno:    Optional[str] = None

    # Datos del taller
    nombre_taller:     str
    direccion:         Optional[str]     = None
    nit:               Optional[str]     = None
    latitud_decimal:   Optional[Decimal] = None
    longitud_decimal:  Optional[Decimal] = None
    tenant_id:         Optional[int]     = None

    class Config:
        from_attributes = True

# -------------------------------------------------------
# INVENTARIO DE TALLER
# -------------------------------------------------------
class InventarioBase(BaseModel):
    item_nombre: str
    cantidad: int

class InventarioCreate(InventarioBase):
    pass

class InventarioOut(InventarioBase):
    id_inventario: int
    taller_id: int

    class Config:
        from_attributes = True